moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -142,18 +142,21 @@ void planning_pipeline::PlanningPipeline::configure()
     if (adapter_plugin_loader_)
       for (const std::string& adapter_plugin_name : adapter_plugin_names_)
       {
-        planning_request_adapter::PlanningRequestAdapterConstPtr ad;
+        planning_request_adapter::PlanningRequestAdapter* ad;
         try
         {
-          ad = adapter_plugin_loader_->createUniqueInstance(adapter_plugin_name);
+          ad = adapter_plugin_loader_->createUnmanagedInstance(adapter_plugin_name);
```

Comment:
Creating unmanaged plugins is very bad practice and needs to be reverted.
As `initialize()` requires a modifyable pointer, just use a non-const `PlanningRequestAdapterPtr` or even better a `UniquePtr`.

Reply:
I think it's fine in this case since either the instance fails to be constructed and the exception is caught or the valid pointer is immediately managed after constructing a PlanningRequestAdapterConstPtr with it. I can change it to `createUniqueInstance()` thought.

Reply:
No, it's not fine at all! When creating the pointer as unmanaged, the _class loader_ cannot keep track of the instance and thus cannot _unload_ the library. This will issue a warning message, which I just recently fixed (#1104).

Reply:
Got it, please review 84e2223